code/game/objects/structures/mouse_cage.dm

```diff
@@ -0,0 +1,130 @@
+
+/obj/item/critter_cage
+	name = "small cage"
+	desc = "A safe place where to keep tiny animals safe. Fit with a drinking bottle that can be refilled."
+	icon = 'icons/obj/virology.dmi'
+	icon_state = "cage"
+	density = 1
+	anchored = 0
+	pressure_resistance = 5
+	flags = FPRINT  | OPENCONTAINER
+
+	var/lock_type = /datum/locking_category/buckle/cage
+
+	var/mob/living/simple_animal/critter = null
+
+
+/obj/item/critter_cage/New()
+	..()
+	create_reagents(10)
+
+/obj/item/critter_cage/Destroy()
+	if (critter)
+		unlock_atom(critter)
+		critter.forceMove(get_turf(src))
+		critter = null
+	..()
+
+/datum/locking_category/buckle/cage
+
+/obj/item/critter_cage/attack_paw(var/mob/user)
+	return attack_hand(user)
+
+/obj/item/critter_cage/attack_hand(var/mob/user)
+	if (critter)
+		if( !user.get_active_hand() )
+			unlock_atom(critter)
+			critter.scoop_up(user)
+			user.visible_message("<span class='notice'>[user] picks up \the [critter].</span>", "<span class='notice'>You pick up \the [critter].</span>")
+			critter = null
+	else
+		MouseDropFrom(user)
+
+/obj/item/critter_cage/attackby(var/obj/O,var/mob/user)
+	. = ..()
+
+	if (isturf(loc))
+		if (!critter && istype (O,/obj/item/weapon/holder/animal))
+			var/obj/item/weapon/holder/animal/store = O
+			var/mob/living/simple_animal/inside = store.stored_mob
+			if (inside.size > SIZE_TINY)
+				to_chat(user, "<span class='warning'>\The [inside] is too big for \the [src]!</span>")
+				return
+			if(!user.drop_item(O, loc))
+				to_chat(user, "<span class='warning'>You can't let go of \the [O]!</span>")
+				return
+			critter = inside
+			qdel(store)
+			critter.forceMove(loc)
+			lock_atom(critter,lock_type)
+
+		if (istype (O,/obj/item/weapon/reagent_containers/food/snacks))
+			if(!user.drop_item(O, loc))
+				to_chat(user, "<span class='warning'>You can't let go of \the [O]!</span>")
+				return
+			O.forceMove(loc)
+			O.pixel_x = pixel_x
+			O.pixel_y = pixel_y-9
+	else
+		to_chat(user, "<span class='warning'>Put the cage down before placing anything inside.</span>")
+
+/obj/item/critter_cage/lock_atom(var/atom/movable/AM)
+	. = ..()
```

Comment:
`lock_atom` can fail, like if the target is already locked, so you need to check for the return value and abort before setting pixel shift.